### PR TITLE
Install libbpf 1.0.1 in buildboxes

### PR DESCRIPTION
**Pull request by jakule**
This is a transition change that only installs new libbpf in our Docker image but still keeps using the "old" version not to break any build. Following PR will make the real switch and remove the "old" libbpf version from our Docker images.

Note: `https://github.com/aquasecurity/libbpfgo` that uses libbpf is not backward compatible, that is why I want to keep libbpf in a separate directory as technically we do support linking with only one version.

### CODE REVIEW

File: build.assets/Dockerfile [hunk 1]
```diff
@@ -56,17 +56,33 @@ RUN git clone --depth=1 https://github.com/Yubico/libfido2.git -b 1.12.0 && \
...
 ARG LIBBPF_VERSION
-RUN mkdir -p /opt && cd /opt && curl -L https://github.com/gravitational/libbpf/archive/refs/tags/v${LIBBPF_VERSION}.tar.gz | tar xz && \
+RUN mkdir -p /opt && cd /opt &&  \
+    curl -L https://github.com/gravitational/libbpf/archive/refs/tags/v${LIBBPF_VERSION}.tar.gz | tar xz && \
     cd /opt/libbpf-${LIBBPF_VERSION}/src && \
+    make &&  \
+    BUILD_STATIC_ONLY=y DESTDIR=/opt/libbpf make install
+
+# Install newer libbpf.
+FROM buildpack-deps:18.04 AS libbpf-1-0-1
```
Comment (codingllama) [inline build.assets/Dockerfile:73]: Use a non-versioned step name, so we don't have to update it when we bump libbpf (and inevitably forget to do so, leading to a confusing step name over time).

Maybe `libbpf-teleport` and `libbpf` (or `libbpf-upstream`)?

File: build.assets/Dockerfile [hunk 2]
```diff
@@ -292,9 +308,9 @@ RUN BIN="/usr/local/bin" && \
       chmod +x "${BIN}/buf"
 
 # Copy BPF libraries.
-COPY --from=libbpf /usr/include/bpf /usr/include/bpf
-COPY --from=libbpf /usr/lib64/ /usr/lib64/
-RUN cd /usr/local/lib && ldconfig
+ARG LIBBPF_VERSION
+COPY --from=libbpf /opt/libbpf/usr /usr/libbpf-${LIBBPF_VERSION}
+COPY --from=libbpf-1-0-1 /opt/libbpf/usr /usr/libbpf-1.0.1
```
Comment (codingllama) [inline build.assets/Dockerfile:313]: Same for a non-versioned path?

File: build.assets/Dockerfile-centos7 [hunk 3]
```diff
@@ -92,6 +92,33 @@ RUN mkdir -p /opt && cd /opt && \
     cd /opt/libbpf-${LIBBPF_VERSION}/src && \
     scl enable devtoolset-11 "make && BUILD_STATIC_ONLY=y DESTDIR=/opt/libbpf make install"
 
+# Install newer libbpf.
+FROM centos:7 AS libbpf-1-0-1
```
Comment (codingllama) [inline build.assets/Dockerfile-centos7:96]: Same as previous comments, here  and in the COPY step.

File: build.assets/Dockerfile-centos7 [hunk 4]
```diff
@@ -92,6 +92,33 @@ RUN mkdir -p /opt && cd /opt && \
...
+    # required by libbpf
+    devtoolset-11-gcc* \
+    # required by libbpf
+    devtoolset-11-make \
+    # required by libbpf
+    elfutils-libelf-devel-static \
+    git \
+    # required by libbpf
+    scl-utils \
+    yum clean all
```
Comment (codingllama) [inline build.assets/Dockerfile-centos7:115]:
```suggestion
    yum install -y \
        git \
        # required by libbpf
        centos-release-scl \
        devtoolset-11-gcc* \
        devtoolset-11-make \
        elfutils-libelf-devel-static \
        scl-utils \
    yum clean all
```

Although we are in a BPF-specific step, so we could do away with all the comments too.

**Review by codingllama — APPROVED** [review]

File: build.assets/Dockerfile (hunk 1 above)
Comment (jakule) [reply build.assets/Dockerfile:73]: @codingllama My plan was to remove this step in my next PR. I don't think we should build two versions of libbpf when we don't really need them.

**Review by codingllama — APPROVED** [review]

**Review by atburke — APPROVED** [review]

**Pull request merged**